src/Panels/ConnectedPlayersPanel.cs

```diff
@@ -0,0 +1,120 @@
+﻿using ColossalFramework.UI;
+using CSM.Helpers;
+using CSM.Networking;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace CSM.Panels
+{
+    public class ConnectedPlayersPanel : UIPanel
+    {
+        private UIButton _closeButton;
+
+        private List<UILabel> _playerLabels;
+        private List<UIButton> _kickButtons;
+
+        private int _playerListLastHash;
+
+        public override void Start()
+        {
+            _playerLabels = new List<UILabel>();
+            _kickButtons = new List<UIButton>();
+
+            // Activates the dragging of the window
+            AddUIComponent(typeof(UIDragHandle));
+
+            backgroundSprite = "GenericPanel";
+            color = new Color32(110, 110, 110, 250);
+
+            // Grab the view for calculating width and height of game
+            UIView view = UIView.GetAView();
+
+            // Center this window in the game
+            relativePosition = new Vector3(view.fixedWidth / 2.0f - 180.0f, view.fixedHeight / 2.0f - 240.0f);
+
+            width = 360;
+            height = 445;
+
+            // Title Label
+            UILabel title = this.CreateTitleLabel("Connected Players", new Vector2(80, -20));
+
+            // Close this dialog
+            _closeButton = this.CreateButton("Close", new Vector2(10, -375));
+            _closeButton.eventClick += (component, param) =>
+            {
+                isVisible = false;
+            };
+
+            base.Start();
+        }
+
+        public override void Update()
+        {
+            // Update the list of players and kick buttons if anything has changed
+            if (isVisible && _playerListLastHash != MultiplayerManager.Instance.PlayerList.GetHashCode())
```

Comment:
The hash is not changing when a player joins or leaves. I think the list uses the default hashcode function, so it's just the internal object id (which is not changing when the contents change).
You could instead use the player count and set a changed flag even when the window is invisible and then update it when it's visible again.